R/neatsort.R

```diff
@@ -0,0 +1,186 @@
+#' # Load required libraries
+#' library(mia)
+#' 
+#' # Load the dataset
+#' data(peerj13075)
+#' 
+#' # Agglomerate by Order and transform the data
+#' tse_order <- mergeFeaturesByRank(peerj13075, rank = "order", onRankOnly = TRUE)
+#' tse_order <- transformAssay(tse_order, assay.type = "counts", method="relabundance", MARGIN = "samples", name="relabundance")
+#' tse_order <- transformAssay(tse_order, assay.type="relabundance", method="z", MARGIN = "features", name="z")
+#' z_transformed_data <- assay(tse_order, "z")
+#' 
+#' # Get the top taxa and perform PCA
+#' top_taxa <- getTopFeatures(tse_order, top = 10, assay.type="z")
+#' top_feature_data <- z_transformed_data[top_taxa, ]
+#' pca_results <- prcomp(top_feature_data, scale = TRUE)
+#' scores_pca <- pca_results$x[, 1:2]
+#' 
+#' # Sort by radial theta and subset the transformed data
+#' sorted_order <- neatsort(scores_pca, dimensions = c(1, 2), centering_method = "mean", sorting_order = "ascending")
+#' ordered_transformed_data <- z_transformed_data[sorted_order, ]
+NULL
+
+#' @rdname neatsort
+setGeneric("neatsort", signature = c("x"),
+           function(x, ...)
+               standardGeneric("neatsort"))
+
+
+# Checks if the input arguments for the neatsort function are valid.
+.check_neatsort_args <- function(x, subset, dimensions, centering_method, sorting_order) {
+    # Check data is a matrix
+    if (!is.matrix(x)) {
+        stop("Input data must be a matrix.")
+    }
+    
+    # Check there is sufficient data
+    if (nrow(x) == 0 || ncol(x) == 0) {
+        stop("No data to plot. Matrix must have at least one row and one column.")
+    }
+    
+    # Check subset validity
+    if (!is.null(subset)) {
+        if (is.numeric(subset) && any(subset > nrow(x))) {
+            stop("Subset refers to rows that do not exist in the data.")
+        } else if (is.character(subset) && any(!subset %in% rownames(x))) {
+            stop("Subset refers to row names that do not exist in the data.")
+        } else if (!is.numeric(subset) && !is.character(subset)) {
+            stop("Subset must be a vector of row indices or names.")
+        }
+    }
+    
+    # Check dimensions are valid
+    if (any(dimensions > ncol(x))) {
+        stop("dimensions refer to columns that do not exist in the data.")
+    }
+    
+    # Check dimension vector is of length 2
+    if (length(dimensions) != 2) {
+        stop("Exactly two dimensions must be specified.")
+    }
+    
+    # Check centering_method
+    centering_method <- match.arg(centering_method, c("mean", "median", "none"))
+    
+    # Check sorting_order
+    sorting_order <- match.arg(sorting_order, c("ascending", "descending"))
+    
+    # Check for unique row names
+    if (any(duplicated(rownames(x)))) {
+        stop("Row names of the matrix must be unique.")
+    }
+    
+    # Check for unique column names
+    if (any(duplicated(colnames(x)))) {
+        stop("Column names of the matrix must be unique.")
+    }
+}
```

Comment:
You could add this below the generic methods ( as they are in most of the files)